polaris-react/src/components/TopBar/components/SearchField/SearchField.scss

```diff
@@ -81,7 +81,7 @@ $search-icon-width: calc(#{$icon-size} + var(--p-space-4));
 
   &::placeholder {
     color: var(--p-text);
-    transition: var(--p-duration-150) color var(--p-ease) 33ms;
```

Comment:
This happens so quickly and on such a thin/small font I don't know how much of the motion I perceive. I personally wouldn't mind removing this one. Would love to know other folks throughts.

Reply:
Yes to either removing or 50ms

Reply:
Just for reference to compare to the videos above here's a video of removing the transition:

https://user-images.githubusercontent.com/21976492/161597844-40f4d33e-021e-45e2-9705-42dbe4f24b63.mp4

Reply:
I'm going to vote for removing the transition. With no transition the focus ring and the placeholder text change at the same time. With the 50ms transition I'm noticing that the placeholder text color change lags slightly after the focus ring and it feels a bit disjointed. 

Would love @johanstromqvist input too but realize he's on vacation this week. I'd be happy to move forward with no transition here for now and if Johan thinks otherwise after returning from vacation I can open up another PR to change it.

Reply:
Sounds like a good plan @lgriffee

Reply:
Great assessment @lgriffee! Thx for ping!

TDLR: Agree that it's safe to remove in this instance!

**Re: focus ring**
- Is focus ring necessary when focusing search field with mouse? My assessment is that the focus ring more often create visual noise and distraction when mousing, instead of providing the clarity it's intended to. This makes suppressing the focus ring an important a11y and perceived perf consideration. We should have a documented opinion on when and why we suppress focus ring.
- If we suppress focus ring on the search field, then we don't have a sync problem.
![](https://screenshot.click/11-21-6rc2g-plgn9.gif)

**Re: default approach when replacing hard coded values**
- I think adjusting a value is a _design tweak_, while removing a transition is more of a _redesign_. We should avoid "redesigning" in scenarios that we don't own or have full context of. It's better if respective product team make their own decisions with the context they have of user needs.
  - If this is a decision that we own, then we can safely change or remove it.
  - If it's not ours, then we should opt for minimum impact, i.e. adjust to closest/most appropriate value such as 50ms.
  - If it's not ours but it's harming UX, then we should reach out to the product team and help them make the change while also learning why.

**Re: replacing values that are likely to be covered shortly?**
- Removing a purposefully applied 33ms might be to optimize for the short term and send the wrong message. It might feel like an improvement to token coverage today, while 33ms might actually be covered tomorrow.
- [As we've discussed in the past](https://github.com/Shopify/polaris/issues/4762), 50ms increments is not sufficient. I frequently use sub-50 durations, such as 17 and 33, and want to encourage product teams to do so as well. This is especially important for choreography and making multiple animations feel as one.

Reply:
Great! Thanks @johanstromqvist  for confirming it’s safe to remove in this instance! 🙏🏻 🎉

**Re: focus ring** 

> Is focus ring necessary when focusing search field with mouse?

I'm not sure and would love to get more history/context on it. From quickly looking through the [MDN `:focus-visible` page](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) it does mention the following: 

> [Cognition](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible#cognition)
> It may not be obvious as to why the focus indicator is appearing and disappearing if a person is using mixed forms of input. For users with cognitive concerns, or who are less technologically literate, this lack of consistent behavior for interactive elements may be confusing.

Perhaps this ⬆️ is why we have the focus ring for both mouse and keyboard input? But I'm not sure why our use is not consistent 🤔 (also love your example).

> We should have a documented opinion on when and why we suppress focus ring.

I totally agree with this for consistency and clarity into decision making. 

> If we suppress focus ring on the search field, then we don't have a sync problem.

Do you mean suppressing it only on mouse click? If so then we would still run into the sync problem with keyboard input (unless we don't have transitions with keyboard focus? Not sure what is best practice in terms of keyboard focus). 

**Re: default approach when replacing hard coded values**
These are really great points about replacing/removing values and I love the decision matrix you outlined. Is the SearchField owned by us since an element used across the admin and is not owned by any one specific section of the admin? 

**Re: replacing values that are likely to be covered shortly?**
This is the only instance of the 33ms getting removed (the other instances that were removed we determined wouldn't get triggered) and I'm happy to add an action item to add back the transition as part of our future motion work (I can also link that work to the focus ring questions we had).